src/muz/base/rule_properties.cpp

```diff
@@ -231,11 +231,13 @@ bool rule_properties::check_accessor(app* n) {
         return is_recognizer_base(t);
     };
 
-    
-    for (unsigned i = ut_size; i < t_size; ++i)
-        if (is_recognizer(m_rule->get_tail(i)))
+    for (unsigned i = ut_size; i < t_size; ++i) {
+        auto *tail = m_rule->get_tail(i);
+        if (is_recognizer(tail))
             return true;
-
+        if (m.is_not(tail) && m_dt.is_recognizer(to_app(tail)->get_arg(0)))
```

Comment:
if (m.is_not(tail, ntail) && m_dt.is_recognizer(ntail))
   recognizers.push_back(ntail);

Reply:
These occurrences of recognizers are of course global and apply to all paths.
So when fixing for context sensitivity within paths there appears to be a use for both global and local context awareness. Alternatively, just delay adding global recognizers to the case where hitting !parent. It is a single pass over the tail.